modules/core/shared/src/main/scala/lucuma/core/math/units.scala

```diff
@@ -3,186 +3,170 @@
+  type WattsPerMeter2Arcsec2
+  given DerivedUnit[WattsPerMeter2Arcsec2, WattsPerMeter2 / (ArcSecond ^ 2), "W/m²/arcsec²", "W/m²/arcsec²"] = DerivedUnit()
+  given TypeString[WattsPerMeter2Arcsec2] = TypeString("W_PER_M_SQUARED_PER_ARCSEC_SQUARED")
 
-  type ErgsPerSecondCentimeter2Arcsec2 = ErgsPerSecondCentimeter2 %/ (ArcSecond %^ 2)
-  implicit val strErgsPerSecondCentimeter2Arcsec2 =
-    UnitStringDef[ErgsPerSecondCentimeter2Arcsec2](
-      "erg/s/cm²/arcsec²",
-      "ERG_PER_S_PER_CM_SQUARED_PER_ARCSEC_SQUARED"
-    )
+  type ErgsPerSecondCentimeter2Arcsec2
+  given DerivedUnit[ErgsPerSecondCentimeter2Arcsec2, ErgsPerSecondCentimeter2 / (ArcSecond ^ 2), "erg/s/cm²/arcsec²", "erg/s/cm²/arcsec²"] = DerivedUnit()
+  given TypeString[ErgsPerSecondCentimeter2Arcsec2] =
+    TypeString("ERG_PER_S_PER_CM_SQUARED_PER_ARCSEC_SQUARED")
 
   // PosInt can be converted to Rational exactly
-  implicit def rationalPosIntConverter[U1, U2](implicit
-    cu: ConvertableUnits[U1, U2]
-  ): UnitConverter[PosInt, U1, Rational, U2] =
-    new UnitConverter[PosInt, U1, Rational, U2] {
-      @inline def vcnv(v: PosInt): Rational =
-        cu.coef * v.value
-    }
-
-  // This can build a converter for units that use PosInt but they are exact only
-  // if the coef is more than 1 and whole, i.e. going from Nanometer to Picometer
-  // The reverse is not true, remaining in the PosInt domain we can't ensure we can go from
-  // Picometer to Nanometer without loosing precision
-  // Thus we shouldn't make this implicit by default~
-  def unsafePosIntConverter[U1, U2](implicit
-    cu: ConvertableUnits[U1, U2]
-  ): UnitConverter[PosInt, U1, PosInt, U2] =
-    new UnitConverter[PosInt, U1, PosInt, U2] {
-      @inline def vcnv(v: PosInt): PosInt =
-        // We only allow the conversion if the coef is more than one and exact
-        if (cu.coef.compareToOne > 0 && cu.coef.isWhole)
-          // given the check above this should be positive and the refinement should always succeed
-          refineV[Positive]((cu.coef * v.value).toInt).getOrElse(sys.error(s"Shouldn't happen"))
-        else
-          sys.error(s"Cannot convert exactly with coef ${cu.coef}")
-
-    }
-
-  // Implicit conversions that can be exact as Pico/Nano/Angstrom are multiples of 10
-  implicit val convNP: UnitConverter[PosInt, Nanometer, PosInt, Picometer] =
-    unsafePosIntConverter[Nanometer, Picometer]
-
-  implicit val convAP: UnitConverter[PosInt, Angstrom, PosInt, Picometer] =
-    unsafePosIntConverter[Angstrom, Picometer]
-
-  implicit val convMP: UnitConverter[PosInt, Micrometer, PosInt, Picometer] =
-    unsafePosIntConverter[Micrometer, Picometer]
+  given rationalPosIntConverter: ValueConversion[PosInt, Rational] = Rational(_)
+
+  inline given [UF, UT]: TruncatingUnitConversion[PosInt, UF, UT] = v =>
+    refineV[Positive](coulomb.conversion.standard.unit.ctx_TUC_Int(v))
+      .getOrElse(refineMV[Int, Positive](1))
```

Comment:
@erikerlandson My own poor attempt at trying to add some support for refined ... I imagine you can do some compile-time magic here :)